features/networking/egress-ip.feature

```diff
@@ -232,3 +232,126 @@ Feature: Egress IP related features
+  @admin
+  @destructive
+  Scenario: The egressIP should still work fine after the node or network service restarted
+    Given the valid egress IP is added to the node
+    And I have a project
+    And I have a pod-for-ping in the project
+
+    # add the egress ip to the project
+    Given as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ip %>"]} |
+
+    When I execute on the pod:
+      | curl | -s | --connect-timeout | 5 | ifconfig.me |
+    Then the step should succeed
+    And the output should contain "<%= cb.valid_ip %>"
+
+    # Reboot the node which patched egressIP
+    Given the host is rebooted and I wait it up to 600 seconds to become available
+    When I execute on the pod:
+      | curl | -s | --connect-timeout | 5 | ifconfig.me |
+    Then the step should succeed
+    And the output should contain "<%= cb.valid_ip %>"
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-15998
+  @admin
+  Scenario Outline: Invalid egressIP should not be acceptable
+    Given I select a random node's host
+    When I run the :patch admin command with:
+      | resource      | hostsubnet                     |
+      | resource_name | <%= node.name %>               |
+      | p             | {"egressIPs": <invalid_ip> }   |
+      | type          | merge                          |
+    Then the step should fail
+    And the output should contain "Invalid JSON Patch"
+
+    Examples:
+      | invalid_ip              |
+      | fe80::5054:ff:fedd:3698 |
+      | a.b.c.d                 |
+      | 10.10.10.-1             |
+      | 10.0.0.1/64             |
+      | 10.1.1/24               |
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-25694
+  @admin
+  @destructive
+  Scenario: Random outages with egressIP
+    Given I store the masters in the :nodes clipboard
+    And the valid egress IP is added to the "<%= cb.nodes[0].name %>" node
+    And I have a project
+    And I have a pod-for-ping in the project
+
+    # add the egress ip to the project
+    Given as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ip %>"]} |
+
+    # Access some external host that is not responding
+    When I execute on the pod:
+      | nc | -v | -w2 | 8.8.8.8 | 23 |
+    Then the step should fail
+    And the output should contain "Operation timed out"
+
+    #Check the sdn logs on pod node
+    Given I use the "<%= pod.node_name %>" node
+    When I get the networking components logs of the node since "90s" ago
+    And the output should not contain "may be offline"
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-25640
+  @admin
+  @destructive
+  Scenario: Should be able to access to the service's externalIP with egressIP
+    Given I have a project
+    Given I store the schedulable nodes in the :nodes clipboard
+    And the Internal IP of node "<%= cb.nodes[0].name %>" is stored in the :hostip clipboard
```

Comment:
Will this work on IPv6 dual-stack?  Or maybe all egressIP tests are IPv4 for now?

Reply:
Currently egressIP only works on openshfit-sdn  for IPv4 for now.